File: app/src/main/java/com/gago/david/myland/Adapters/MyLandRecyclerViewAdapter.java
package com.gago.david.myland.Adapters;

import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.net.Uri;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.gago.david.myland.LandFragment.OnListFragmentInteractionListener;
import com.gago.david.myland.LandOpenHelper;
import com.gago.david.myland.Models.LandObject;
import com.gago.david.myland.Models.PriorityObject;
import com.gago.david.myland.Models.TaskTypeObject;
import com.gago.david.myland.R;

import java.util.List;

import de.hdodenhof.circleimageview.CircleImageView;

/**
 * {@link RecyclerView.Adapter} that can display a {@link LandObject} and makes a call to the
 * specified {@link OnListFragmentInteractionListener}.
 * TODO: Replace the implementation with code for your data type.
 */
public class MyLandRecyclerViewAdapter extends RecyclerView.Adapter<MyLandRecyclerViewAdapter.ViewHolder> {

    private final List<LandObject> mValues;
    private final OnListFragmentInteractionListener mListener;
    private final List<PriorityObject> priorities;

    public MyLandRecyclerViewAdapter(List<LandObject> items, OnListFragmentInteractionListener listener, List<PriorityObject> priorities) {
        mValues = items;
        mListener = listener;
        this.priorities = priorities;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.fragment_land, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(final ViewHolder holder, int position) {
        holder.mItem = mValues.get(position);
        holder.mIdView.setText(holder.mItem.name);
        holder.mContentView.setText(holder.mItem.Description);
        holder.landImage.setImageBitmap(new LandOpenHelper((Context) mListener).getImage(holder.mItem.imageUri));
        holder.number.setText(String.format("%d",holder.mItem.notifications));
        boolean colorChanged = false;
        for (PriorityObject p : priorities)
            if(p.p_order == holder.mItem.priority) {
                holder.notification.setImageDrawable(new ColorDrawable(Color.parseColor(p.color)));
                colorChanged = true;
                break;
            }
        if(!colorChanged)
            holder.notification.setImageDrawable(new ColorDrawable(Color.WHITE));

        holder.mView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (null != mListener) {
                    // Notify the active callbacks interface (the activity, if the
                    // fragment is attached to one) that an item has been selected.
                    mListener.onListFragmentInteraction(holder.mItem);
                }
            }
        });
    }

    @Override
    public int getItemCount() {
        return mValues.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        public final View mView;
        public final TextView mIdView;
        public final TextView mContentView;
        public final CircleImageView landImage;
        public final CircleImageView notification;
        public final TextView number;
        public LandObject mItem;

        public ViewHolder(View view) {
            super(view);
            mView = view;
            mIdView = view.findViewById(R.id.id);
            mContentView = view.findViewById(R.id.content);
            landImage = view.findViewById(R.id.land_image);
            notification = view.findViewById(R.id.notification);
            number = view.findViewById(R.id.number);
        }

        @Override
        public String toString() {
            return super.toString() + " '" + mContentView.getText() + "'";
        }
    }
}

File: app/src/main/java/com/gago/david/myland/adapters/MyLandRecyclerViewAdapter.kt
package com.gago.david.myland.adapters

import android.content.Context
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.gago.david.myland.LandFragment
import com.gago.david.myland.LandOpenHelper
import com.gago.david.myland.R
import com.gago.david.myland.models.LandObject
import com.gago.david.myland.models.PriorityObject
import com.google.android.material.progressindicator.LinearProgressIndicator
import de.hdodenhof.circleimageview.CircleImageView

class MyLandRecyclerViewAdapter(
    private val mValues: List<LandObject>,
    private val mListener: LandFragment.OnListFragmentInteractionListener?,
    private val priorities: List<PriorityObject>,
    private val emptyView: View,
    private val context: Context
) : RecyclerView.Adapter<MyLandRecyclerViewAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.land_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.mItem = mValues[position]
        holder.mIdView.text = holder.mItem!!.name
        holder.mContentView.text = holder.mItem!!.description
        holder.landImage.setImageBitmap(LandOpenHelper.getImage(context, holder.mItem!!.imageUri))
        holder.number.text = String.format("%d", holder.mItem!!.notifications)
        var colorChanged = false
        for (p in priorities) if (p.p_order == holder.mItem!!.priority) {
            holder.notification.setImageDrawable(ColorDrawable(Color.parseColor(p.color)))
            colorChanged = true
            break
        }
        if (!colorChanged) holder.notification.setImageDrawable(ColorDrawable(Color.WHITE))
        val progress = (holder.mItem!!.totalTasks - holder.mItem!!.notifications.toFloat()) / holder.mItem!!.totalTasks.coerceAtLeast(1) * 100
        holder.progressBar.setProgress(progress.toInt(), true)
        holder.mView.setOnClickListener { mListener?.onListFragmentInteraction(holder.mItem!!) }
    }

    override fun getItemCount(): Int {
        if (mValues.isEmpty())
            emptyView.visibility = View.VISIBLE
        else
            emptyView.visibility = View.GONE
        return mValues.size
    }

    inner class ViewHolder(val mView: View) : RecyclerView.ViewHolder(mView) {
        val mIdView: TextView = mView.findViewById(R.id.id)
        val mContentView: TextView = mView.findViewById(R.id.content)
        val landImage: CircleImageView = mView.findViewById(R.id.land_image)
        val notification: CircleImageView = mView.findViewById(R.id.notification)
        val number: TextView = mView.findViewById(R.id.number)
        val progressBar: LinearProgressIndicator = mView.findViewById(R.id.land_progress)
        var mItem: LandObject? = null
        override fun toString(): String {
            return super.toString() + " '" + mContentView.text + "'"
        }

    }
}